### PR TITLE
[MAINTENANCE] Unset cloud vars when running pytest from a mac

### PR DESCRIPTION
**Pull request by kenwade4**

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import os
```
Comment (alexsherstinsky) [inline tests/__init__.py:1]: @cdkini @anthonyburdi @billdirks @Kilo59 -- Something tells me that OSS should not have any mentions of `GE_CLOUD` or other non-OSS type branding (similar sentiment as not including internal JIRA ticket links in the OSS code), but wanted to know what you think.  Thanks. /cc @Shinnnyshinshin

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import sys
+
+if sys.platform == "darwin":
+    print(
+        "Running tests from a dev machine, so unsetting any GE_CLOUD_* environment vars"
+    )
+    cloud_vars = [k for k in os.environ if k.startswith("GE_CLOUD")]
+    for var_name in cloud_vars:
+        print(f"Unsetting environment variable {var_name}")
+        del os.environ[var_name]
```
Comment (cdkini) [inline tests/__init__.py]: I think we should probably discuss this a bit more before implementing.

We've got both `GE_` and `GX_` env vars to consider

**Review by cdkini — CHANGES_REQUESTED** [review]

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import os
+import sys
+
+if sys.platform == "darwin":
+    cloud_vars = [k for k in os.environ if k[:8] in ("GX_CLOUD", "GE_CLOUD")]
```
Comment (cdkini) [inline tests/__init__.py]: Could we instead use the `GXCloudEnvironmentVariable` enum as a check here?

File: tests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    if cloud_vars:
+        print(
+            "Running tests from a dev machine, so unsetting any CLOUD environment vars"
+        )
+        for var_name in cloud_vars:
+            print(f"Unsetting environment variable {var_name}")
+            del os.environ[var_name]
```
Comment (cdkini) [inline tests/__init__.py]:
```suggestion
if sys.platform == "darwin":
    for var in GXCloudEnvironmentVariable:
        if var in os.environ:
            logger.info(...) # or print
            os.environ.pop(var)
```

**Review by cdkini — APPROVED** [review]

**Pull request merged**